backend/Alpaki.Logic/Features/Invitations/RegisterVolunteer/RegisterVolunteerHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Alpaki.CrossCutting.Enums;
+using Alpaki.Database;
+using Alpaki.Database.Models;
+using Alpaki.Logic.Services;
+using MediatR;
+using Microsoft.AspNet.Identity;
+
+namespace Alpaki.Logic.Features.Invitations.RegisterVolunteer
+{
+    public class RegisterVolunteerHandler : IRequestHandler<RegisterVolunteer, RegisterVolunteerResponse>
+    {
+        private readonly IDatabaseContext _dBContext;
+        private readonly JwtGenerator _jwtGenerator;
+        private readonly IPasswordHasher _passwordHasher;
+
+        public RegisterVolunteerHandler(IDatabaseContext dBContext, JwtGenerator jwtGenerator)
+        {
+            _dBContext = dBContext;
+            _jwtGenerator = jwtGenerator;
+            _passwordHasher = new PasswordHasher();
+        }
+        public async Task<RegisterVolunteerResponse> Handle(RegisterVolunteer request, CancellationToken cancellationToken)
+        {
+            var invitation = _dBContext.Invitations.SingleOrDefault(
```

Comment:
Async

Reply:
dodaj uniq index na email. jeśli jakimś cudem będą dwa takie same wpisy z takim samym mailem to single weźmie pierwszy. a jeśli wolontariuesz będzie miał kod do drugiego zaproszenia to nigdy nie będzie w stanie przejść procesu. poza tym ja bym robił selecta po kodzie! Jeśli kod jest błędny to koniec przygody!